examples/allocation/tinygo/testdata/greet.go

```diff
@@ -63,32 +64,24 @@ func _greeting(ptr, size uint32) (ptrSize uint64) {
 // ptrToString returns a string from WebAssembly compatible numeric types
 // representing its pointer and length.
 func ptrToString(ptr uint32, size uint32) string {
-	// Get a slice view of the underlying bytes in the stream. We use SliceHeader, not StringHeader
-	// as it allows us to fix the capacity to what was allocated.
-	return *(*string)(unsafe.Pointer(&reflect.SliceHeader{
-		Data: uintptr(ptr),
-		Len:  uintptr(size), // Tinygo requires these as uintptrs even if they are int fields.
-		Cap:  uintptr(size), // ^^ See https://github.com/tinygo-org/tinygo/issues/1284
-	}))
+	return unsafe.String((*byte)(unsafe.Pointer(uintptr(ptr))), size)
 }
 
 // stringToPtr returns a pointer and size pair for the given string in a way
 // compatible with WebAssembly numeric types.
+// The returned pointer aliases the string hence the string must be kept alive
+// until ptr is no longer needed.
 func stringToPtr(s string) (uint32, uint32) {
-	buf := []byte(s)
-	ptr := &buf[0]
-	unsafePtr := uintptr(unsafe.Pointer(ptr))
-	return uint32(unsafePtr), uint32(len(buf))
+	ptr := unsafe.Pointer(unsafe.StringData(s))
```

Comment:
summary of a chat with @aykevl on this is that even though StringData can return nil the latter case to uintptr is safe. Special casing `len(s)` zero to return early with 0, 0 is both unnecessary and may slow down the code. Plus we'd have to clarify that empty isn't always ptr 0, for example a result of a function that reslices a non-empty string to zero length.